thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java

```diff
@@ -1075,19 +1078,29 @@ public Response getAlertFilterMTTD(@PathParam("functionId") @NotNull long id,
       @QueryParam("severity") @NotNull double severity) {
     AnomalyFunctionDTO anomalyFunctionSpec = anomalyFunctionDAO.findById(id);
     BaseAlertFilter alertFilter = alertFilterFactory.fromSpec(anomalyFunctionSpec.getAlertFilter());
-    // Compute minimum-time-to-detect for both pattern, and choose the minimum as the minimum-time-to-detect for alert filter
-    // For one pattern alert, the MTTD is always +Infinity for the other pattern, hence by taking minimum the evaluate results will lie in the pattern of interest
-    // For two pattern alerts, the MTTD is computed to take as the minimum value of both side
-    double alertFilterMTTDInHour = Double.POSITIVE_INFINITY;
-    if (!Double.isNaN(alertFilter.getAlertFilterMTTD(severity))) {
-      alertFilterMTTDInHour = Math.min(alertFilterMTTDInHour, alertFilter.getAlertFilterMTTD(severity));
+    // Compute minimum-time-to-detect for both pattern, and choose the maximum value as the minimum-time-to-detect for alert filter
+    // For one pattern alert, the MTTD is always +Infinity for the other pattern, hence this step we only use MTTD from the interested pattern
+    // For two pattern alerts, the MTTD is computed to take as the maximum value of both side
+    double alertFilterMTTDInHour = 0;
+    double alertFilterMTTDUP = alertFilter.getAlertFilterMTTD(severity);
+    double alertFilterMTTDDOWN = alertFilter.getAlertFilterMTTD(-1.0 * severity);
+    if (!Double.isInfinite(alertFilterMTTDUP) && !Double.isNaN(alertFilterMTTDUP)) {
+      alertFilterMTTDInHour = Math.max(alertFilterMTTDInHour, alertFilterMTTDUP);
     }
-    if (!Double.isNaN(alertFilter.getAlertFilterMTTD(-1.0 * severity))) {
-      alertFilterMTTDInHour = Math.min(alertFilterMTTDInHour, alertFilter.getAlertFilterMTTD(-1.0 * severity));
+    if (!Double.isInfinite(alertFilterMTTDDOWN) && !Double.isNaN(alertFilterMTTDDOWN) ) {
+      alertFilterMTTDInHour = Math.max(alertFilterMTTDInHour, alertFilterMTTDDOWN);
     }
+
     TimeUnit detectionUnit = anomalyFunctionSpec.getBucketUnit();
     int detectionBucketSize = anomalyFunctionSpec.getBucketSize();
-    double functionMTTDInHour = TimeUnit.HOURS.convert(detectionBucketSize, detectionUnit);
+    int detectionWindowSize = 1;
+    try {
+      Properties functionProps = toProperties(anomalyFunctionSpec.getProperties());
+      detectionWindowSize = Integer.valueOf(functionProps.getProperty(SIGN_TEST_WINDOW_SIZE, "1"));
+    } catch (IOException e) {
+      LOG.warn("Failed to fetch function properties when evaluating mttd!");
+    }
+    double functionMTTDInHour = TimeUnit.HOURS.convert(detectionBucketSize * detectionWindowSize, detectionUnit);
```

Comment:
For sign test, the detection window is minute, but the window size is 6 hours. The calculation is wrong here.

Reply:
This is not using "window size" for Sign Test as the function. The mttd is based on minimum detection window, which is based on "signTestWindowSize" as the key from Sign Test.